src/wrapper/objects/jinputstream.rs

```diff
@@ -0,0 +1,86 @@
+use crate::{
+    errors::*,
+    objects::{AutoLocal, JMethodID, JObject},
+    signature::{JavaType, Primitive},
+    sys::jsize,
+    JNIEnv,
+};
+use std::io;
+
+const DEFAULT_BUF_SIZE: usize = 8 * 1024; // same as std::io
+
+/// Wrapper for JObjects that implement `java.io.InputStream`.
+pub struct JInputStream<'a: 'b, 'b> {
+    internal: JObject<'a>,
+    buffer: AutoLocal<'a, 'b>,
+    read: JMethodID<'a>,
+    env: &'b JNIEnv<'a>,
+}
+
+impl<'a: 'b, 'b> ::std::ops::Deref for JInputStream<'a, 'b> {
```

Comment:
style nit: In modern Rust, I can drop `::` at the beginning of the import, so use `std::ops::Deref`.

Reply:
Done